nncf/common/pruning/utils.py

```diff
@@ -396,16 +394,14 @@ def calculate_in_out_channels_by_masks(pruning_groups: List[Cluster[PrunedLayerI
     and updating corresponding input channels number in `pruning_groups_next_nodes` nodes.
 
     :param pruning_groups: A list of pruning groups.
-    :param masks: A dictionary of masks of each pruning node.
-    :param tensor_processor: NNCF Tensor processor to operate on NNCFTensors.
+    :param num_of_sparse_elements_by_node: A dictionary of num_of_sparse_elements of each pruning node.
     :param full_input_channels:  A dictionary of input channels number in original model.
     :param full_output_channels: A dictionary of output channels number in original model.
     :param pruning_groups_next_nodes: A dictionary of next nodes of each pruning group.
     :return Dictionary of new input channels number {node_name: channels_num}
     """
     def get_num_of_sparse_elements_by_node(node_name: str) -> int:
-        mask = masks[node_name]
-        return mask.shape[0] - int(tensor_processor.sum(mask))
+        return num_of_sparse_elements_by_node[node_name]
```

Comment:
You can pass the `num_of_sparse_elements_by_node` dictionary to `_calculate_in_out_channels` as is.

Reply:
And move code from `_calculate_in_out_channels` to `calculate_in_out_channels_by_masks` and rename it to `calculate_in_out_channels`.

Reply:
There is another method `calculate_in_out_channels_in_uniformly_pruned_model` which is using `_calculate_in_out_channels` with its custom sparse element counter.